tests/next_tests/unit_tests/test_constructors.py

```diff
@@ -57,50 +99,48 @@ def test_zeros(allocator, device):
         allocator=allocator,
         device=device,
     )
-    ref = np.zeros((sizes[I], sizes[J])).astype(gtx.float32)
+    ref = xp.zeros((sizes[I], sizes[J])).astype(gtx.float32)
 
-    assert np.array_equal(a.ndarray, ref)
+    assert xp.array_equal(a.ndarray, ref)
 
 
-# TODO: parametrize with gpu backend and compare with cupy array
-@pytest.mark.parametrize(
-    "allocator, device",
-    [
-        [next_allocators.StandardCPUFieldBufferAllocator(), None],
-        [None, core_defs.Device(core_defs.DeviceType.CPU, 0)],
-    ],
-)
-def test_ones(allocator, device):
+def test_ones(allocator_device_refnamespace):
+    allocator, device, xp = allocator_device_refnamespace
     a = gtx.ones(
         common.Domain(dims=(I, J), ranges=(common.UnitRange(0, 10), common.UnitRange(0, 10))),
         dtype=core_defs.dtype(np.float32),
         allocator=allocator,
         device=device,
     )
-    ref = np.ones((sizes[I], sizes[J])).astype(gtx.float32)
+    ref = xp.ones((sizes[I], sizes[J])).astype(gtx.float32)
 
-    assert np.array_equal(a.ndarray, ref)
+    assert xp.array_equal(a.ndarray, ref)
 
 
-# TODO: parametrize with gpu backend and compare with cupy array
-@pytest.mark.parametrize(
-    "allocator, device",
-    [
-        [next_allocators.StandardCPUFieldBufferAllocator(), None],
-        [None, core_defs.Device(core_defs.DeviceType.CPU, 0)],
-    ],
-)
-def test_full(allocator, device):
+def test_full(allocator_device_refnamespace):
+    allocator, device, xp = allocator_device_refnamespace
     a = gtx.full(
         domain={I: range(sizes[I] - 2), J: (sizes[J] - 2)},
         fill_value=42.0,
         dtype=core_defs.dtype(np.float32),
         allocator=allocator,
         device=device,
     )
-    ref = np.full((sizes[I] - 2, sizes[J] - 2), 42.0).astype(gtx.float32)
+    ref = xp.full((sizes[I] - 2, sizes[J] - 2), 42.0).astype(gtx.float32)
+
+    assert xp.array_equal(a.ndarray, ref)
 
-    assert np.array_equal(a.ndarray, ref)
+
+def test_copy():
```

Comment:
Isn't this test the same than the one in the embedded `test_nd_array_field` tests?